doc/neps/nep-0031-uarray.rst

```diff
@@ -61,7 +71,8 @@ called ``unumpy`` `[8]`_, the implementation of which is already underway. Again
 will not be explained here, the reader should refer to its documentation for this purpose.
 
 The only change this NEP proposes at its acceptance, is to make ``unumpy`` the officially recommended
-way to override NumPy. ``unumpy`` will remain a separate repository/package, and will be developed
+way to override NumPy. ``unumpy`` will remain a separate repository/package (which we propose to vendor
+rather than depend on for the time being), and will be developed
```

Comment:
How about a soft dependency, like in `scipy.fft`?

Reply:
That will probably be fine, just thinking about if we should put that in here. The key thing is vendoring. Whether to optionally override the vendored version with an installed version to be able to ship bug fixes quicker is a small detail.

Note also that it's not a normal soft dependency - that typically means that not installed == no functionality.

I'd actually leave it out, since it's not important at this point.

Reply:
>Note also that it's not a normal soft dependency - that typically means that not installed == no functionality.

Is there an accepted term for this? I was just looking for a word between hard and optional.

Either way, the "optionally overriding" behaviour is very important in order for the vendored version to play nicely with other libraries. For example, if I have a decorator `@implements(unumpy.sum)` such as in NEP 18, then my backend *won't work at all* if my `unumpy.sum` is not the same as the user's `unumpy.sum`. This is exactly the problem that we get through normal vendoring.

Reply:
> Is there an accepted term for this? 

Not that I know of - I think it requires a one sentence description.

> then my backend _won't work at all_ if my `unumpy.sum` is not the same as the user's `unumpy.sum`

that's a good point. isn't an issue for `scipy.fft` (right?) but is tricky here. I think the soft dependency solves the issue, as long as other libraries don't do the same. If they either import from the numpy vendored package, or from an installed `unumpy` directly that should both work.

Reply:
>isn't an issue for scipy.fft (right?) 

Yes, this is only an issue for `numpy` because `unumpy` is the package defining the multimethods. In `scipy.fft` we own the multimethods ourself so these cannot be mismatched, even if the `uarray` libraries were.

>I think the soft dependency solves the issue

Completely agree, I just wanted to point out that shipping bug fixes faster is not the only motivation.